### PR TITLE
BUG-106826 new authorization domain model introduced

### PR DESCRIPTION
**Pull request by attilapalfi92**

### CODE REVIEW

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/security/User.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    @Convert(converter = JsonToString.class)
+    @Column(columnDefinition = "TEXT", name = "tenant_permissions", nullable = false)
+    private Json tenantPermissions;
+
+    @ManyToOne
+    private Tenant tenant;
+
+    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
+    private Set<UserOrgPermissions> organizations;
```
Comment (akanto) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/security/User.java]: field has an incorrect name

Comment (attilapalfi92) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/security/User.java]: fixed

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/security/Tenant.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    @SequenceGenerator(name = "tenant_generator", sequenceName = "tenant_id_seq", allocationSize = 1)
+    private Long id;
+
+    @Basic(optional = false)
+    private String name;
+
+    @OneToMany(mappedBy = "tenant", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
+    private Set<User> users = new HashSet<>();
+
+    @OneToMany(mappedBy = "tenant", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
```
Comment (bbihari) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/security/Tenant.java]: LAZY would be better

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/security/User.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    private Long id;
+
+    private String name;
+
+    @Basic(optional = false)
+    private String email;
+
+    private String company;
+
+    private String idp;
```
Comment (bbihari) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/security/User.java]: We don't know what to write into this field, I think it's better if we delete it for now

**Pull request merged**